scripts/Muon/GUI/Common/load_widget/load_presenter.py

```diff
@@ -88,4 +89,10 @@ def unreg_on_loading_finished(self, slot):
         self.view.unreg_on_loading_finished(slot)
 
     def get_run_num_loaded_detectors(self, run):
-        return self.load_model.num_loaded_detectors[run]
+        num_detectors = 0
+        try:
+            group = mantid.mtd[run]
+            num_detectors = len(group)
+        except:
```

Comment:
Is it possible to have a more specific exception here ?

Reply:
Is this a `TypeError`?

Reply:
```suggestion
        except KeyError:
```